satpy/readers/__init__.py

```diff
@@ -370,7 +370,7 @@ def get_valid_reader_names(reader):
     return new_readers
 
 
-def available_readers(as_dict=False):
+def available_readers(as_dict=False, yaml_loader=UnsafeLoader):
```

Comment:
Is there a reason to use the unsafe loader here or would it just work with the BaseLoader?

Reply:
Well I was just setting the default here to be "backwards" compatible meaning leave the old default functionality unchanged as much as possible. I don't know what the intentions are. If this is set to BaseLoader as a default this will always list all readers with yaml config files even if they are missing dependencies. So if a user uses this to find out which readers are available to him this might be irritating if he then tries one that can't work due to missing dependencies. If this is changed it, at least, needs to be documented somewhere. if this has other implications I didn't check.

Reply:
Good point.

Reply:
I thought the BaseLoader raised an error when it encountered python imports and other "dynamic" YAML features. This makes sense to me for backwards compatibility and to keep the dependency checking the default.